### ISSUE TITLE
Some basic seeds are not beatable with proposed logic

### ISSUE THREAD

**Issue by minous27** [minous27/TheMessengerRandomizerMod#22]
After the 0.3 release, there were reports of a couple of seeds that were generated but did not honor the logic. This issue will track them and the work to fill the logic gaps found.
[16384_log.txt](https://github.com/minous27/TheMessengerRandomizerMod/files/5336179/16384_log.txt)
[18093_log.txt](https://github.com/minous27/TheMessengerRandomizerMod/files/5336181/18093_log.txt)

**Comment by minous27** [comment]
Notes for seed 18093 - At some point grapple gets blocked by magic boots which gets blocked by fairy/crown for courage check. Wingsuit blocks fairy bottle but wingsuit doesnt know that it blocks magic boots because of that. I need to look closer into why fairy bottle doesnt know that it is blocking magic boots.
If I can fill that disconnect then when wingsuit tries to get placed at Sun crest it will fail because it is blocking magic boots.

**Comment by minous27** [comment]
Thoughts from the most recent findings:

> Hmm...So wingsuit got placed first, so tabi is actaully the blocker
But it doesnt know that because the other requirements (crown and fairy) havent been placed yet.
So there is the flaw
The issue is the order the items are being placed. When things happen too close together then things dont know their interdependencies.
Just gotta figure out how to handle that.

[23724_log.txt](https://github.com/minous27/TheMessengerRandomizerMod/files/5461040/23724_log.txt)

**Comment by minous27** [comment]
One solution could be to place each item requirement one at a time...That might ensure that dependencies are considered each time...feels inefficient tho...

**Comment by minous27** [comment]
Some changes have been made to prevent seeds from being generated if they are not beatable. However, this causes alot of attempts for a good seed and if it hits a threshold then it just gives it the bad seed anyway to prevent freezing. Still have some work to do to help with this.

**Comment by minous27** [comment]
Seed generation logic has been removed from the mod and is now handled by a [separate UI](https://github.com/minous27/TheMessengerRandoGenSite). While building that site I was able to fix a few bugs that would cause this issue. If it pops up again I'll open a new issue in the site's github.
